### PR TITLE
Convert the "Bug report" template to YAML form

### PR DESCRIPTION
**Pull request by dshevtsov**
### Description (*)

Convert the Bug report issue from markdown type to YAML forms.

### Fixed Issues (if relevant)

1. Fixes magento-commerce/community-guild#61

### Manual testing scenarios (*)

To preview the new template, [open it in my fork](https://github.com/dshevtsov/magento2/blob/ds_issue-forms/.github/ISSUE_TEMPLATE/bug_report.yaml).

[Use Case Example in this comment](https://github.com/magento/magento2/pull/35103#issuecomment-1113368620)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by sdzhepa** [comment]
@dshevtsov 

I would propose to change orders for these files and some naming according to usual bug reports standards

1. Environment and Preconditions (not required)
2. Steps to Reproduce *
3. Expected Result *
4. Actual Results*
5. Additional Information (not required)
6. Release Notes (not required)

**Comment by sdzhepa** [comment]
@dshevtsov 
Sorry for the very short statement without any argumentation 
let me clarify  my comment and provide more explanations

And yes probably "best practices" is more suitable that "standards"
Also, i mean only sections related to
```
- Environment and Preconditions (not required)
- Steps to Reproduce *
- Expected Result *
- Actual Results*
```
The "Additional Information (not required)" field is based on my experience working on the Magento Open Source GitHub track. This section could contain links to the comments with important information or Root Cuse analysis or additional video recordings etc. anything that important for the issue and by some reason cannot be added to the main sections



1. I referred to practices that the Magento team used internally(~10 years) to report bug reports(regular bug, sprint bugs, regression bugs). A couple of links from internal wikis
[](url)
-  [[Regression DUMMY BUG] Functionality is behaving incorrectly](https://jira.corp.magento.com/browse/MC-42081) (used as a template for GL Regression Testing teams)
- [DOD(Extension Quality Platform and Services (EQPS))](https://wiki.corp.adobe.com/display/EQPS2/Definitions+of+Done#DefinitionsofDone-Backlog-IssueCreation) 
- Marketplace [Bug Report Guidelines](https://wiki.corp.magento.com/display/MARKETPLACE/Bug+Report+Guidelines)](https://wiki.corp.magento.com/display/MARKETPLACE/Bug+Report+Guidelines)
- Marketplace [Bug Reporting](https://wiki.corp.magento.com/display/MARKETPLACE/QA+Process#QAProcess-BugReporting)
- [MCOM "How to report a Bug"](https://wiki.corp.magento.com/display/EEI/How+to+report+a+Bug)

2. This internal "standard" originally was used as requirements for issue reporting on Magento OS public GitHub and was described in ["Report an issue"](https://devdocs.magento.com/contributor-guide/contributing.html#report) on DevDocs 

3. Unfortunately ISO and RUP standards/templates for bug/defect reports aro not available for free but additionally can be googled simple request [bug report structure](https://www.google.com/search?q=bug+report+structure&rlz=1C1GCEA_enDE812IE813&sxsrf=APq-WBvMwk2eg1vurfnlaEhn7sWy0L7F-g%3A1645637195460&ei=S24WYu6-G4_EytMPiJioqAk&oq=bug+report+srtucture&gs_lcp=Cgdnd3Mtd2l6EAMYADIECAAQDToECCMQJzoLCC4QgAQQxwEQowI6CwguEIAEEMcBENEDOgUIABCABDoOCC4QgAQQxwEQowIQ1AI6BAgAEEM6CwguEIAEEMcBEK8BOgoIABCABBCHAhAUOgUIABDLAToFCCEQoAE6BwghEAoQoAFKBAhBGABKBAhGGABQAFj4G2DZJmgAcAF4AIABzAGIAYcSkgEGNC4xNS4xmAEAoAEBwAEB&sclient=gws-wiz)

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      description: |
+        Tell us what you expected to happen.
+      placeholder: |
+        Example:
+          Order is placed successfully, customer is redirected to the success page.
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Current result
```
Comment (sdzhepa) [inline .github/ISSUE_TEMPLATE/bug_report.yaml]: I would propose to change the "Current result" to the "Actual Result"
to have consistency with the [previous template](https://github.com/magento/magento2/blame/2.4-develop/.github/ISSUE_TEMPLATE/bug_report.md#L31) and internal Bug report standards

**Review by sdzhepa — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+name: Bug report
+description: Technical issue with the Magento 2 core components
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Please read [our guidelines](https://devdocs.magento.com/contributor-guide/contributing.html#report) before submitting the issue.
```
Comment (sdzhepa) [inline .github/ISSUE_TEMPLATE/bug_report.yaml]: Looks like the page was moved https://developer.adobe.com/commerce/contributor/guides/code-contributions/#report-an-issue

**Review by sdzhepa — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  - type: textarea
+    attributes:
+      label: Release note
+      description: |
+        Help us to provide meaningful release notes to the community.
+  - type: checkboxes
+    attributes:
+      label: Triage and priority
+      description: |
+        Provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as a Reporter.
```
Comment (sdzhepa) [inline .github/ISSUE_TEMPLATE/bug_report.yaml]: Page is moved https://developer.adobe.com/commerce/contributor/guides/code-contributions/#community-backlog-priority

**Review by sdzhepa — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 2 above)
Comment (dshevtsov) [reply .github/ISSUE_TEMPLATE/bug_report.yaml]: good catch

Comment (dshevtsov) [inline .github/ISSUE_TEMPLATE/bug_report.yaml]:
```suggestion
        Please read [our guidelines (https://developer.adobe.com/commerce/contributor/guides/code-contributions/) before submitting the issue.
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 3 above)
Comment (dshevtsov) [reply .github/ISSUE_TEMPLATE/bug_report.yaml]: fixed

**Review by sdzhepa — APPROVED** [review]

**Pull request merged**